mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/EntityInjectorGenerator.java

```diff
@@ -0,0 +1,115 @@
+  private final String targetParameterName;
+
+  public EntityInjectorGenerator(
+      ExecutableElement methodElement,
+      DaoImplementationGenerator daoImplementationGenerator,
+      ProcessorContext context) {
+    this.methodElement = methodElement;
+    this.daoImplementationGenerator = daoImplementationGenerator;
+    // We're expecting a void method, where one parameter is an annotated entity, and the other a
+    // SettableByName or subtype
+    TypeMirror returnType = methodElement.getReturnType();
+    if (returnType.getKind() != TypeKind.VOID || methodElement.getParameters().size() != 2) {
+      context
+          .getMessager()
+          .error(
+              methodElement,
+              "%s methods must have two parameters and a void return type",
+              EntityInjector.class.getSimpleName());
+      throw new SkipGenerationException();
+    }
+    String tmpEntity = null;
+    TypeElement tmpEntityElement = null;
+    String tmpTarget = null;
+    for (VariableElement parameterElement : methodElement.getParameters()) {
+      TypeMirror parameterType = parameterElement.asType();
+      if (context.getClassUtils().implementsSettableByName(parameterType)) {
+        tmpTarget = parameterElement.getSimpleName().toString();
+      } else if (parameterType.getKind() == TypeKind.DECLARED) {
+        Element parameterTypeElement = ((DeclaredType) parameterType).asElement();
+        if (parameterTypeElement.getKind() == ElementKind.CLASS
+            && parameterTypeElement.getAnnotation(Entity.class) != null) {
+          tmpEntity = parameterElement.getSimpleName().toString();
+          tmpEntityElement = ((TypeElement) parameterTypeElement);
+        }
+      }
+    }
+    if (tmpEntity == null || tmpTarget == null) {
+      context
+          .getMessager()
+          .error(
+              methodElement,
+              "Could not match parameters, expected a SettableByName "
+                  + "and an annotated entity (in any order)");
+      throw new SkipGenerationException();
+    }
+    this.entityParameterName = tmpEntity;
+    this.entityElement = tmpEntityElement;
+    this.targetParameterName = tmpTarget;
+  }
+
+  @Override
+  public void addMembers(TypeSpec.Builder classBuilder) {
+    String helperFieldName = daoImplementationGenerator.addEntityHelperField(entityElement);
+
+    MethodSpec.Builder overridingMethodBuilder =
+        MethodSpec.methodBuilder(methodElement.getSimpleName().toString())
+            .addAnnotation(Override.class)
+            .addModifiers(Modifier.PUBLIC);
+    for (VariableElement parameterElement : methodElement.getParameters()) {
+      overridingMethodBuilder.addParameter(
+          ClassName.get(parameterElement.asType()), parameterElement.getSimpleName().toString());
+    }
+    // Forward to the base injector in the helper:
+    overridingMethodBuilder.addStatement(
+        "$L.inject($L, $L)", helperFieldName, entityParameterName, targetParameterName);
```

Comment:
what the `inject` call means? We are using some already existing code?

Reply:
Yeah that's a bit hard to follow, it's code from another generated class.

For each `@Entity`, the processor now generates an implementation of `EntityHelper` (in mapper-runtime), that defines `inject()`.
Upon encountering the entity, this class requests that the generated DAO code create an instance of this helper, and store it in a field (line 94):
```java
String helperFieldName = daoImplementationGenerator.addEntityHelperField(entityElement);
```
Then it generates this line of code that delegates to this helper field.

Here's what the generated code looks like:
```java
public class UserDao_Impl implements UserDao {
  private final MapperContext context;

  private final User_Helper userHelper; // The helper field

  private UserDao_Impl(MapperContext context) {
    this.context = context;
    this.userHelper = new User_Helper(context);
  }

  // This is the part that's generated by EntityInjectorGenerator:
  @Override
  public void inject(User user, BoundStatement boundStatement) {
    userHelper.inject(user, boundStatement); // delegate to the helper
  }

  ... // (init methods, not relevant here)
}
```